apps/dashr-lastodash/app.R

```diff
@@ -0,0 +1,379 @@
+}
+
+get_plot_list <- function(nplots) {
+  lapply(seq_len(nplots), function(x) plot_ly())
+}
+
+generate_curves <- function(height = 950,
+                            width = 800,
+                            bg_color = "white",
+                            font_size = 10,
+                            tick_font_size = 8,
+                            line_width = 0.5) {
+  yval <- "DEPT"
+  xvals <- list(
+    list("BTVPVS", "DGRC"),
+    list("R15P", "R09P", "R39P", "R27P", "EWXT"),
+    list("ALCDLC", "ALDCLC"),
+    list("TNPS"),
+    list("BTCSS", "BTCS")
+  )
+  plots <- list()
+  for (i in 1:length(xvals)) {
+    ifelse(i == 2, line_style <- 'dashdot', line_style <- 'solid')
+    p <- plot_ly(las_data$A, y = las_data$A[[yval]])
+    for (xval in xvals[[i]]) {
+      p <- add_trace(p, x = las_data$A[[xval]], name = xval, mode = 'lines', type = "scatter",
+                     line = list(dash = line_style, width = line_width))
+    }
+    plots[length(plots)+1] <- p
+  }
+  fig <- subplot(
+    plots[[1]], plots[[2]], plots[[3]], plots[[4]], plots[[5]], 
+    get_plot_list(5), nrows = 1, shareY = TRUE, margin = 0
+  )
+  fig$x$data[[2]][["xaxis"]] = "x6"
+  fig$x$data[[7]][["xaxis"]] = "x7"
+  fig$x$data[[9]][["xaxis"]] = "x8"
+  fig$x$data[[12]][["xaxis"]] = "x9"
+  for (i in 1:length(xvals)) {
+    for (xval in xvals[[i]]) {
+      ifelse(i == 1, xaxis <- "xaxis", xaxis <- glue("xaxis{i}"))
+      ifelse(i == 2, fig$x$layout[[xaxis]]$type <- "log", fig$x$layout[[xaxis]]$type <- "linear")
+    }
+  }
+  for (i in c(6:9)) {
+    xaxis <- glue("xaxis{i}")
+    ifelse(i != 9, fig$x$layout[[xaxis]]$overlaying <- glue("x{i-5}"), fig$x$layout[[xaxis]]$overlaying <- glue("x{i-4}"))
+    fig$x$layout[[xaxis]]$anchor <- "y"
+    fig$x$layout[[xaxis]]$side <- "top"
+    fig$x$layout[[xaxis]]$title <- glue("TITLE {i}")
+  }
+  for (i in 1:9) {
+    ifelse(i %in% c(1:5), key <- xvals[[i]][[1]], ifelse(i !=  9, key <- xvals[[i-5]][[length(xvals[[i-5]])]], key <- xvals[[i-4]][[length(xvals[[i-4]])]]))
+    ifelse(i == 1, xaxis <- "xaxis", xaxis <- glue("xaxis{i}"))
+    fig$x$layout[[xaxis]]$mirror <- "all"
+    fig$x$layout[[xaxis]]$automargin <- TRUE
+    fig$x$layout[[xaxis]]$showline <- TRUE
+    fig$x$layout[[xaxis]]$title <- list(text = generate_axis_title(las_data$C, key), font = list(family = "Arial, sans-serif", size = font_size))
+    fig$x$layout[[xaxis]]$tickfont <- list(family = "Arial, sans-serif", size = tick_font_size)
+  }
+  # y-axis
+  fig$x$layout[["yaxis"]]$mirror <- "all"
+  fig$x$layout[["yaxis"]]$automargin <- TRUE
+  fig$x$layout[["yaxis"]]$showline <- TRUE
+  fig$x$layout[["yaxis"]]$title <- list(text = generate_axis_title(las_data$C, "DEPT"), font = list(family = "Arial, sans-serif", size = font_size))
+  fig$x$layout[["yaxis"]]$tickfont <- list(family = "Arial, sans-serif", size = tick_font_size)
+  fig$x$layout[["yaxis"]]$autorange <- 'reversed'
+  # layout
+  fig$x$layout$height <- height
+  fig$x$layout$width <- width
```

Comment:
I had that because it would increase the width of the table in the printable pdf. Maybe I can just create a callback to allow resizing when the print button is clicked since I agree, a wider graph would be more preferable in the app itself.

Reply:
Looks OK as is;  we can revisit if needed later, but the UI and features are 👌 for now!